src/mqt/predictor/rl/cost_model.py

```
@@ -0,0 +1,280 @@
+
+    def add_gate_to_cost_table(gate: str, qubits: int, params: int, controls: int) -> None:
+        total_qubits = qubits + controls
+        qc = QuantumCircuit(total_qubits)
+        try:
+            gate_name = "c" * controls + gate if controls > 0 else gate
+            if params == 0:
+                getattr(qc, gate_name)(*list(range(total_qubits)))
+            else:
+                param_values = list(range(1, params + 1))
+                getattr(qc, gate_name)(*param_values, *list(range(total_qubits)))
+        except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)